ios/Classes/PTPluginUtils.m

```diff
@@ -20,35 +58,459 @@ + (NSNumber *)PT_idAsNSNumber:(id)value
+    } else {
+        flutterResult(resultString);
     }
 }
 
 
-+ (void)importAnnotationCommand:(NSString *)xfdfCommand documentViewController:(PTDocumentViewController *)docVC
++ (void)flattenAnnotations:(bool)formsOnly resultToken:(FlutterResult)flutterResult documentViewController:(PTDocumentViewController *)docVC
 {
+    
+    [docVC.toolManager changeTool:[PTPanTool class]];
+    
     if(docVC.document == Nil)
     {
         // something is wrong, no document.
         NSLog(@"Error: The document view controller has no document.");
+        
+        flutterResult([FlutterError errorWithCode:@"flatten_annotations" message:@"Failed to flatten annotations" details:@"Error: The document view controller has no document."]);
+        return;
+    }
+    
+    NSError *error;
+    
+    [docVC.pdfViewCtrl DocLock:YES withBlock:^(PTPDFDoc * _Nullable doc) {
+        [doc FlattenAnnotations:formsOnly];
+    } error:&error];
+    
+    if(error)
+    {
+        NSLog(@"Error: Failed to flatten annotations from doc. %@", error.localizedDescription);
+        flutterResult([FlutterError errorWithCode:@"flatten_annotations" message:@"Failed to flatten annotations" details:@"Error: Failed to flatten annotations from doc."]);
+    } else {
+        flutterResult(nil);
+    }
+}
+
++ (void)deleteAnnotations:(NSString *)annotationList resultToken:(FlutterResult)flutterResult documentViewController:(PTDocumentViewController *)docVC
+{
+    if(docVC.document == Nil)
+    {
+        // something is wrong, no document.
+        NSLog(@"Error: The document view controller has no document.");
+
+        flutterResult([FlutterError errorWithCode:@"delete_annotations" message:@"Failed to delete annotations" details:@"Error: The document view controller has no document."]);
+        return;
+    }
+    
+    NSError* error;
+    
+    NSArray *annotArray = [PTPluginUtils PT_idAsArray:[PTPluginUtils PT_JSONStringToId:annotationList]];
+    
+    NSArray* matchingAnnots = [PTPluginUtils findAnnotsWithUniqueIDs:annotArray documentViewController:docVC error:&error];
+    
+    if (error) {
+        NSLog(@"Error: Failed to get annotations from doc. %@", error.localizedDescription);
+        
+        flutterResult([FlutterError errorWithCode:@"delete_annotations" message:@"Failed to delete annotations" details:@"Error: Failed to get annotations from doc."]);
+        return;
+    }
+    
+    for (PTAnnot *annot in matchingAnnots) {
+        [docVC.pdfViewCtrl DocLock:YES withBlock:^(PTPDFDoc * _Nullable doc) {
+            PTPage *page = [annot GetPage];
+            if (page && [page IsValid]) {
+                int pageNumber = [page GetIndex];
+                [docVC.toolManager willRemoveAnnotation:annot onPageNumber:pageNumber];
+                
+                [page AnnotRemoveWithAnnot:annot];
+                [docVC.pdfViewCtrl UpdateWithAnnot:annot page_num:pageNumber];
```

Comment:
We can't use `UpdateWithAnnot:` after deleting the annot reliably. Instead you can just call `Update:YES` after the loop to refresh everything. (If you want to get fancy you can get the annot's rect before deleting it, and then update rect. But if you have more than 2-3 annots, I'd recommend updating the entire thing anyway because calling Update again and again in side the loop will be slow.)